From f176a8b5a637c0d45a647c80b9dfe698fc2fa736 Mon Sep 17 00:00:00 2001
From: Peter Vingelmann <petya@master.com>
Date: Wed, 11 May 2016 03:05:04 +0200
Subject: [PATCH 3/5] Fix unused warnings in boost/asio/error.hpp

---
 boost/asio/error.hpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/boost/asio/error.hpp b/boost/asio/error.hpp
index 4252c15..89fc80e 100644
--- a/boost/asio/error.hpp
+++ b/boost/asio/error.hpp
@@ -255,15 +255,6 @@ inline const boost::system::error_category& get_addrinfo_category()
 extern BOOST_ASIO_DECL
 const boost::system::error_category& get_misc_category();
 
-static const boost::system::error_category& system_category
-  = boost::asio::error::get_system_category();
-static const boost::system::error_category& netdb_category
-  = boost::asio::error::get_netdb_category();
-static const boost::system::error_category& addrinfo_category
-  = boost::asio::error::get_addrinfo_category();
-static const boost::system::error_category& misc_category
-  = boost::asio::error::get_misc_category();
-
 } // namespace error
 } // namespace asio
 } // namespace boost
-- 
2.8.0.windows.1

