From b624fd2fe7a2caec7b6f42853de514f732c2edf4 Mon Sep 17 00:00:00 2001
From: Peter Vingelmann <petya@master.com>
Date: Fri, 12 Oct 2018 23:52:58 +0200
Subject: [PATCH] Ignore deprecated register warnings for Python 2.7 headers

---
 boost/python/detail/wrap_python.hpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/boost/python/detail/wrap_python.hpp b/boost/python/detail/wrap_python.hpp
index e2b7c185..d4068a8f 100644
--- a/boost/python/detail/wrap_python.hpp
+++ b/boost/python/detail/wrap_python.hpp
@@ -156,6 +156,19 @@ typedef int pid_t;
 #if PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION == 2 && PY_MICRO_VERSION < 2
 # include <boost/python/detail/python22_fixed.h>
 #else
+// Some header files for Python 2.7 use the "register" specifier which is
+// deprecated in C++14 and removed in C++17. Ignore these warnings in newer
+// compilers to maintain compatibility with Python 2.7.
+# if defined(BOOST_GCC)
+#  if BOOST_GCC >= 70000
+#     pragma GCC diagnostic ignored "-Wregister"
+#  endif
+# elif defined(__clang__) && defined(__has_warning)
+#  if __has_warning("-Wdeprecated-register")
+#     pragma GCC diagnostic ignored "-Wdeprecated-register"
+#  endif
+# endif
+
 # include <Python.h>
 #endif
 
-- 
2.18.0.windows.1

