diff --git a/boost/atomic/detail/platform.hpp b/boost/atomic/detail/platform.hpp
index 3dfb73d..ff1848b 100644
--- a/boost/atomic/detail/platform.hpp
+++ b/boost/atomic/detail/platform.hpp
@@ -16,8 +16,9 @@
 #endif

 // Intel compiler does not support __atomic* intrinsics properly, although defines them (tested with 13.0.1 and 13.1.1 on Linux)
-#if (defined(__GNUC__) && ((__GNUC__ * 100 + __GNUC_MINOR__) >= 407) && !defined(BOOST_INTEL_CXX_VERSION))\
-    || (defined(BOOST_CLANG) && ((__clang_major__ * 100 + __clang_minor__) >= 302))
+#if ((defined(__GNUC__) && ((__GNUC__ * 100 + __GNUC_MINOR__) >= 407) && !defined(BOOST_INTEL_CXX_VERSION))\
+    || (defined(BOOST_CLANG) && ((__clang_major__ * 100 + __clang_minor__) >= 302)))\
+    && (defined(__GCC_ATOMIC_BOOL_LOCK_FREE) && (__GCC_ATOMIC_BOOL_LOCK_FREE == 2))

     #include <boost/atomic/detail/gcc-atomic.hpp>

